gitclone/gitclone_test.go

```diff
@@ -0,0 +1,411 @@
+package gitclone
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/bitrise-io/bitrise-init/step"
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/command/git"
+)
+
+const always = 9999
+const rawCmdError = "dummy_cmd_error"
+
+func Test_checkoutState(t *testing.T) {
+	tests := []struct {
+		name       string
+		cfg        Config
+		cmdOutputs map[string]commandOutput
+		want       *step.Error
+		wantCmds   []string
+	}{
+		{
+			name: "Checkout commit",
+			cfg: Config{
+				RepositoryURL: "https://github.com/bitrise-io/steps-git-clone.git",
```

Comment:
Could you move the url and the commit to a constant?

Reply:
I think excessive use of constants makes the tests harder to read.

Reply:
I would agree with you if the actual value of repoURL and commit had an impact on the test outcome, but this is not the case, at least not in every test. By extracting them to constants you actually make the test less cluttered by only exposing values that are important for that particular test case.

Reply:
removed repo URL where not needed.